src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/functional/legalrep/SubmitClaimTest.java

```diff
@@ -0,0 +1,100 @@
+
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+
+public class SubmitClaimTest extends BaseLegalRepTest {
+
+    private User solicitor;
+
+    @Before
+    public void before() {
+        solicitor = idamTestService.createSolicitor();
+    }
+
+    @Test
+    public void shouldSuccessfullySubmitClaimDataAndReturnCreatedCase() {
+        ClaimData claimData = testData.submittedBySolicitorBuilder().build();
+
+        Claim createdCase = submitClaim(claimData).then()
+                                                  .statusCode(HttpStatus.OK.value())
+                                                  .and()
+                                                  .extract()
+                                                  .body()
+                                                  .as(Claim.class);
+
+        assertThat(claimData).isEqualTo(createdCase.getClaimData());
+        assertThat(createdCase.getCreatedAt()).isCloseTo(LocalDateTimeFactory.nowInLocalZone(),
+                                                         within(2, ChronoUnit.MINUTES));
+    }
+
+    @Test
+    public void shouldReturnUnprocessableEntityWhenInvalidClaimIsSubmitted() {
+        ClaimData invalidClaimData = testData.submittedBySolicitorBuilder().withAmount(null).build();
+
+        submitClaim(invalidClaimData).then().statusCode(HttpStatus.UNPROCESSABLE_ENTITY.value());
+    }
+
+    @Test
+    public void shouldReturnConflictResponseWhenClaimDataWithDuplicatedExternalIdIsSubmitted() {
+        UUID externalId = UUID.randomUUID();
+
+        ClaimData claimData = testData.submittedBySolicitorBuilder().withExternalId(externalId).build();
+
+        submitClaim(claimData).andReturn();
+        submitClaim(claimData).then().statusCode(HttpStatus.CONFLICT.value());
+    }
+
+    @Test
+    public void shouldReturnUnprocessableEntityWhenClaimWithInvalidTimelineIsSubmitted() {
+        ClaimData invalidClaimData = testData.submittedBySolicitorBuilder()
+                                             .withTimeline(SampleTimeline.builder()
+                                                                         .withEvents(asList(new TimelineEvent[1001]))
+                                                                         .build())
+                                             .build();
+
+        submitClaim(invalidClaimData).then().statusCode(HttpStatus.UNPROCESSABLE_ENTITY.value());
+    }
+
+    @Test
+    public void shouldReturnUnprocessableEntityWhenClaimWithInvalidEvidenceIsSubmitted() {
+        ClaimData invalidClaimData = testData.submittedBySolicitorBuilder()
+                                             .withEvidence(SampleEvidence.builder()
+                                                                         .withRows(asList(new EvidenceRow[1001]))
```

Comment:
1001 is too many rows? Why don't you add a const with self-descriptive name?

Reply:
Note all of the `/legalrep/*` tests are just copies from the originals in `/citizen/*` with relevant substitutions.

When I do this for real I will definitely fix this magic number though